src/components/SplitCsvUpload/SplitCsvUpload.tsx

```diff
@@ -0,0 +1,104 @@
+import { UploadOutlined } from '@ant-design/icons'
+import { Trans } from '@lingui/macro'
+import TooltipIcon from 'components/TooltipIcon'
+import { ThemeContext } from 'contexts/themeContext'
+import { Split } from 'models/v2/splits'
+import { ChangeEventHandler, useContext } from 'react'
+import { splitPercentFrom } from 'utils/v2/math'
+
+// https://developer.mozilla.org/en-US/docs/Web/API/FileReader/readyState
+const READY_STATE_DONE = 2
+
+const readCsvFile = async (file: File): Promise<string | null> => {
+  const reader = new FileReader()
+
+  return new Promise((resolve, reject) => {
```

Comment:
should these functions be typed?

Reply:
Hmm, types are inferred in this case because it's specified in the function return (line 12). Happy to add them explicitly if you feel strongly 🙏 

![Uploading Screen Shot 2022-08-09 at 12.11.05 PM.png…]()